app/scripts/modules/amazon/src/function/configure/FunctionTags.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+
+import { IWizardPageComponent, MapEditor } from '@spinnaker/core';
+import { FormikProps } from 'formik';
+import { IAmazonFunctionUpsertCommand } from 'amazon/index';
+import { IAmazonFunction } from 'amazon/domain';
+
+export interface IFunctionTagsProps {
+  formik: FormikProps<IAmazonFunctionUpsertCommand>;
+  isNew?: boolean;
+  functionDef: IAmazonFunction;
+}
+
+export interface IFunctionTagsState {
+  some: string;
+}
+
+export class FunctionTags extends React.Component<IFunctionTagsProps, IFunctionTagsState>
+  implements IWizardPageComponent<IAmazonFunctionUpsertCommand> {
+  private duplicateKeys = false;
+
+  public validate = () => {
+    const errors = {} as any;
+
+    if (this.duplicateKeys) {
+      errors.vars = 'Tags have duplicate keys.';
+    }
+    return errors;
+  };
+
+  public componentDidMount() {
```

Comment:
Same comment about doing this by assigning state in the constructor.

Reply:
done. will appear in the next push.